### PR TITLE
Create smaller container image

### PR DESCRIPTION
**Pull request by dza89**

### CODE REVIEW

**Comment by dza89** [comment]
Shrinks the image from 990 to 9.8 mb

File: Dockerfile [hunk 1]
```diff
@@ -1,10 +1,23 @@
-FROM golang:1.13-buster
+FROM golang:1.13-buster as build
+
+RUN apt update && apt install apt ca-certificates && \
```
Comment (luca-sd) [inline Dockerfile]: I think it'd be better to use `apt-get` with this kind of scripts as `apt` is meant to be used interactively, is there a specific reason to prefer it?

File: Dockerfile [hunk 2]
```diff
@@ -1,10 +1,23 @@
...
+    useradd -u 1000 ekscloudwatch
 
 WORKDIR /app
 
 COPY go.mod ./
 RUN go mod download
 COPY . .
-RUN go build -o ekscloudwatch github.com/sysdiglabs/ekscloudwatch/cmd
+RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 go build -ldflags="-s -w" -o ekscloudwatch github.com/sysdiglabs/ekscloudwatch/cmd && \
```
Comment (luca-sd) [inline Dockerfile]: since we use the absolute path `/app/ekscloudwatch` everywhere we could put it in the `-o` arg as well

**Review by luca-sd — COMMENTED** [review]
Thank you for this PR, looks great! Refactoring of the Dockerfile was definitely needed, especially the user and leaner image, now it can be pulled instantly :) I only have a couple minor comments.

File: Dockerfile (hunk 2 above)
Comment (dza89) [reply Dockerfile]: agreed and fixed

File: Dockerfile (hunk 1 above)
Comment (dza89) [reply Dockerfile]: Completely correct, didn't pay attention. Fixed

File: Dockerfile [hunk 3]
```diff
@@ -1,10 +1,23 @@
-FROM golang:1.13-buster
+FROM golang:1.13-buster as build
+
+RUN apt-get update && apt-get install apt ca-certificates && \
```
Comment (luca-sd) [inline Dockerfile]: awesome! nit: seems to build right, but maybe a `-y` wouldn't hurt

**Comment by dza89** [comment]
@luca-sd 
Fixed

**Comment by dza89** [comment]
@luca-sd 
Can you put the new docker image in https://hub.docker.com/r/sysdiglabs/k8sauditlogforwarder ?

**Comment by luca-sd** [comment]
The version is now updated in the deployment. Thanks again! :)

**Pull request merged**